Actions/AL-Go-Helper.ps1

```diff
@@ -258,70 +280,102 @@ function DownloadAndImportBcContainerHelper {
                 $bcContainerHelperVersion = "preview"
                 Write-Host "Download failed, downloading BcContainerHelper $bcContainerHelperVersion version from Blob Storage"
                 $webclient.DownloadFile("https://bccontainerhelper.blob.core.windows.net/public/$($bcContainerHelperVersion).zip", "$tempName.zip")
             }
         }
-        elseif ($bcContainerHelperVersion -eq "preview" -or $bcContainerHelperVersion -eq "dev") {
-            # For backwards compatibility, use preview when dev is specified
-            Write-Host "Downloading BcContainerHelper $bcContainerHelperVersion version from Blob Storage"
-            $webclient.DownloadFile("https://bccontainerhelper.blob.core.windows.net/public/preview.zip", "$tempName.zip")
-        }
         else {
-            Write-Host "Downloading BcContainerHelper $bcContainerHelperVersion version from CDN"
-            $webclient.DownloadFile("https://bccontainerhelper.azureedge.net/public/$($bcContainerHelperVersion).zip", "$tempName.zip")
+            $tempName = Join-Path $bcContainerHelperRootFolder ([Guid]::NewGuid().ToString())
+            if ($bcContainerHelperVersion -eq "preview" -or $bcContainerHelperVersion -eq "dev") {
+                # For backwards compatibility, use preview when dev is specified
+                Write-Host "Downloading BcContainerHelper $bcContainerHelperVersion version from Blob Storage"
+                $webclient.DownloadFile("https://bccontainerhelper.blob.core.windows.net/public/preview.zip", "$tempName.zip")
+            }
+            else {
+                Write-Host "Downloading BcContainerHelper $bcContainerHelperVersion version from CDN"
+                $webclient.DownloadFile("https://bccontainerhelper.azureedge.net/public/$($bcContainerHelperVersion).zip", "$tempName.zip")
+            }
         }
         Expand-7zipArchive -Path "$tempName.zip" -DestinationPath $tempName
-        $BcContainerHelperPath = (Get-Item -Path (Join-Path $tempName "*\BcContainerHelper.ps1")).FullName
+        $bcContainerHelperPath = (Get-Item -Path (Join-Path $tempName "*\BcContainerHelper.ps1")).FullName
         Remove-Item -Path "$tempName.zip" -ErrorAction SilentlyContinue
-    }
-    . $BcContainerHelperPath @params
-    $tempName
-}
-
-function CleanupAfterBcContainerHelper {
-    Param(
-        [string] $bcContainerHelperPath
-    )
-
-    if ($bcContainerHelperPath) {
-        try {
-            Write-Host "Removing BcContainerHelper"
-            Remove-Module BcContainerHelper
-            Remove-Item $bcContainerHelperPath -Recurse -Force
+        if ($bcContainerHelperVersion -notlike "https://*") {
+            # Check whether the version is already available in the cache
+            $version = Get-Content -Encoding UTF8 -Path (Join-Path $tempName 'BcContainerHelper/Version.txt')
+            $cacheFolder = Join-Path $bcContainerHelperRootFolder $version
```

Comment:
Correct - today, it has to download to get the version and use the cache.
Cache is only used on self-hosted runners - on GitHub hosted runners, the cache will always be empty.
Only reason for the cache now is to not fill up temp space with 1000s of similar versions of BcContainerHelper - and avoid plastering all workflows with cleanup steps.
When we move the BcContainerHelper to download from NuGet, we will be able to avoid the download as well.